### ISSUE TITLE
很好的工作，但是... Good work, but ...

### ISSUE THREAD

**Issue by hijkzzz** [marlbenchmark/on-policy#2]
如果你们让QMIX用8个进程，增大Batch Size和或者每次训练epoch次数，最后加上 TD(lambda)<=0.5
QMIX能把这些算法干趴下
参考我们的简略调参 https://arxiv.org/abs/2102.03479
我真的发现 MARL这个领域 由于调参的问题导致一大堆错误的结论和实验甚至motivation出发就错了，涉及十篇+ CCFA顶会paper
尤其是 AAAI 这个会议的文章连证明是错的都能 accept

**Comment by hijkzzz** [comment]
English Version

If you let QMIX also use 128 processes (8 is fine), increase the Batch Size and or the number of epochs per training, and finally add TD(lambda)<=0.5
QMIX can take these algorithms down
Refer to our abbreviated tuning reference https://arxiv.org/abs/2102.03479
I really found that MARL is a field where the problem of parametrization leads to a lot of wrong conclusions and experiments and even motivation is wrong from the beginning, involving 10+ CCFA top papers

**Comment by jxwuyi** [comment]
Thanks for the comments.

We are not surprised that with better tuning, QMix can be even better. However, this is parallel to our focus. What our paper shows is that policy gradient algorithms, which are typically assumed to be less sample efficient, can be surprisingly effective compared with the off-policy algorithms, which are used the most in the existing MARL literature.

Regarding the wrong conclusions and motivations, as one of the initial contributors of deep MARL literature, I partially agree that some of the conclusions might worth re-examinination but disagree that they are all wrong. It is possible that the execution procedure of research ideas can be improved but strong evidence would be expected before making such claims. I would encourage the authors to carefully study the existing literature and analyze what specifically is problematic to benefit the community.

Here is a really nice example of how researchers from Google did so in a scientific manner previously: https://arxiv.org/abs/1802.10031

**Comment by TonghanWang** [comment]
> English Version
> 
> If you let QMIX also use 8 processes, Increase the Batch Size and or the number of epochs per training, and finally add TD(lambda)<=0.5
> QMIX can beat all these algorithms.
> Refer to our simple tuning: RIIT https://arxiv.org/abs/2102.03479
> I really found that MARL is a field where the problem of hyperparameters leads to a lot of wrong conclusions and experimental results and even the motivation is wrong from the beginning, involving 10+ CCFA top papers


Hi, thanks for your search about the implementation tricks. However, I think more experimental results and theoretical analyses are expected to support your strong conclusion.

Most of the papers studied in your work are not motivated by tasks in the SMAC benchmark. Some SMAC tasks do not require a high degree of coordination among agents and typically do not suffer from problems like relative overgeneralization. For example, on the map corridor, independent PPO has good performance. On more complex tasks like predator and prey, theoretically and empirically, algorithms like WQMIX and QPLEX would have better performance. 

Therefore, depending on several maps from the SMAC benchmark without analyses of the task structure may not benefit the MARL community. More experiments characterized by unique challenges of multi-agent learning would be expected for a more beneficial discussion.

**Comment by hijkzzz** [comment]
> > English Version
> > If you let QMIX also use 8 processes, Increase the Batch Size and or the number of epochs per training, and finally add TD(lambda)<=0.5
> > QMIX can beat all these algorithms.
> > Refer to our simple tuning: RIIT https://arxiv.org/abs/2102.03479
> > I really found that MARL is a field where the problem of hyperparameters leads to a lot of wrong conclusions and experimental results and even the motivation is wrong from the beginning, involving 10+ CCFA top papers
> 
> Hi, thanks for your search about the implementation tricks. However, I think more experimental results and theoretical analyses are expected to support your strong conclusion.
> 
> Most of the papers studied in your work are not motivated by tasks in the SMAC benchmark. Some SMAC tasks do not require a high degree of coordination among agents and typically do not suffer from problems like relative overgeneralization. For example, on the map corridor, independent PPO has good performance. On more complex tasks like predator and prey, theoretically and empirically, algorithms like WQMIX and QPLEX would have better performance.
> 
> Therefore, results on several maps from the SMAC benchmark without analyses of the task structure may not benefit the MARL community. More experiments characterized by unique challenges of multi-agent learning would be expected for a more beneficial discussion.

都是中国人，那我直接用中文回答吧。假设 Predator Prey 比 SMAC 更复杂吧，很明显这个 Task 更适合用连续控制来做（参考 MADDPG）。
我直接用 Deep Multi-Agent Reinforcement Learning for Decentralised Continuous Cooperative Control 里面的实验结果说明。
这里使用的 Predator-Prey 是完全合作性质的，如下图所示，所以干掉单调性约束性能增强了？

![image](https://user-images.githubusercontent.com/19810594/110295008-8ea41880-802b-11eb-965f-a5b960e954e9.png)

**Comment by hijkzzz** [comment]
> Thanks for the comments.
> 
> We are not surprised that with better tuning, QMix can be even better. However, this is parallel to our focus. What our paper shows is that policy gradient algorithms, which are typically assumed to be less sample efficient, can be surprisingly effective compared with the off-policy algorithms, which are used the most in the existing MARL literature.
> 
> Regarding the wrong conclusions and motivations, as one of the initial contributors of deep MARL literature, I partially agree that some of the conclusions might worth re-examinination but disagree that they are all wrong. It is possible that the execution procedure of research ideas can be improved but strong evidence would be expected before making such claims. I would encourage the authors to carefully study the existing literature and analyze what specifically is problematic to benefit the community.
> 
> Here is a really nice example of how researchers from Google did so in a scientific manner previously: https://arxiv.org/abs/1802.10031

吴教授，您好
我觉得您的这份工作很好，提出一些存在的问题。
谢谢
